js/oop/Factorial.js

```diff
@@ -2,26 +2,72 @@ import { parseNumber } from "../helper.js";
 
 export class Factorial {
   /** @type {number} */
-  n;
+  number;
 
   /**
-   * @param {number} n
+   * @param {number} number
    */
-  constructor(n) {
-    this.n = parseNumber(n);
+  constructor(number) {
+    this.number = parseNumber(number);
   }
 
-  // write your code here
+  /**
+   * Count factorial number from the given "n" value using loop way.
+   *
+   */
+  countUsingLoop() {
+    var result = this.number <= 0 ? 1 : this.number;
+
+    for (let i = 1; i < this.number; i++) {
+      result = result * (this.number - i);
+    }
+
+    return result;
+  }
+
+  /**
+   * Count factorial number from the given "n" value using recursive way.
+   *
+   * @param {number | undefined} n
+   */
+  countUsingRecursive(number = undefined) {
+    if (number == undefined) {
+      return 1;
+    }
+
+    if (number < 2) {
+      return 1;
+    } else {
+      return number * this.countUsingRecursive(number - 1);
+    }
+  }
+
+  count(method) {
+    let result;
+
+    if (method === "loop") {
+      result = this.countUsingLoop();
+    } else if (method === "recursive") {
+      result = this.countUsingRecursive(this.number);
+    } else {
+      throw new Error("Method must be loop or recursive.");
+    }
+
+    return result;
+  }
 }
 
 document.getElementById("form").addEventListener("submit", function (event) {
   event.preventDefault();
 
   try {
-    const n = event.target["n"].value;
+    const number = event.target["n"].value;
     const method = event.target["method"].value;
 
-    const result = new Factorial(n);
+    console.log(number);
+    console.log(method);
```

Comment:
It's very dangerous to use any console method in the production environment because anyone can see what happens inside your code. So please don't use it unless you're debugging your code